email_parser.py

```diff
@@ -0,0 +1,28 @@
+import sys
+from utils.email_detector import EmailDetector
+from utils.http_handler import HttpHandler
+from utils.website_crawler import WebsiteCrawler
+
+if len(sys.argv) != 2:
+    print "Usage: $ email_parser HTTP_URL"
+    sys.exit()
+
+emails = set()
+
+website_url = sys.argv[1]
+http_handler = HttpHandler(website_url)
+html_response = http_handler.raw_response
+website_crawler = WebsiteCrawler(html_response, website_url)
+website_crawler.feed(html_response)
+
+for website in website_crawler.get_websites():
+    # TODO: Implement multithreading
```

Comment:
We could try pre-processing the website, so we only handle tags that we're interested in, like removing all the JavaScript code, or the CSS styles.
Could also try to do dynamic programming, saving the websites/info we've already gotten from previous crawls.